Library/Homebrew/cmd/install-vendor.sh

```diff
@@ -0,0 +1,166 @@
+  then
+    curl_args="-fLA"
+  elif [[ -z "$HOMEBREW_QUIET" ]]
+  then
+    curl_args="-#fLA"
+  else
+    curl_args="-sfLA"
+  fi
+
+  temporary_path="${CACHED_LOCATION}.incomplete"
+
+  mkdir -p "$HOMEBREW_CACHE"
+  [[ -n "$HOMEBREW_QUIET" ]] || echo "==> Downloading $VENDOR_URL"
+  if [[ -f "$CACHED_LOCATION" ]]
+  then
+    [[ -n "$HOMEBREW_QUIET" ]] || echo "Already downloaded: $CACHED_LOCATION"
+  else
+    if [[ -f "$temporary_path" ]]
+    then
+      "$HOMEBREW_CURL" "$curl_args" "$HOMEBREW_USER_AGENT_CURL" \
+                       -C - "$VENDOR_URL" -o "$temporary_path"
+      if [[ $? -eq 33 ]]
+      then
+        [[ -n "$HOMEBREW_QUIET" ]] || echo "Trying a full download"
+        rm -f "$temporary_path"
+        "$HOMEBREW_CURL" "$curl_args" "$HOMEBREW_USER_AGENT_CURL" \
+                         "$VENDOR_URL" -o "$temporary_path"
+      fi
+    else
+      "$HOMEBREW_CURL" "$curl_args" "$HOMEBREW_USER_AGENT_CURL" \
+                       "$VENDOR_URL" -o "$temporary_path"
+    fi
+
+    trap '' SIGINT
+    mv "$temporary_path" "$CACHED_LOCATION"
+    trap - SIGINT
+  fi
+
+  if [[ -n "$(which shasum)" ]]
+  then
+    sha="$(shasum -a 256 "$CACHED_LOCATION" | cut -d' ' -f1)"
+  elif [[ -n "$(which sha256sum)" ]]
+  then
+    sha="$(sha256sum "$CACHED_LOCATION" | cut -d' ' -f1)"
+  else
+    odie "Cannot verify the checksum ('shasum' or 'sha256sum' not found)!"
+  fi
+
+  if [[ "$sha" != "$VENDOR_SHA" ]]
+  then
+    odie <<EOS
+Checksum mismatch.
+Expected: $VENDOR_SHA
+Actual: $sha
+Archive: $CACHED_LOCATION
+To retry an incomplete download, remove the file above.
+EOS
+  fi
+}
+
+install() {
+  local tar_args
+
+  if [[ -n "$HOMEBREW_VERBOSE" ]]
+  then
+    tar_args="xvzf"
+  else
+    tar_args="xzf"
+  fi
+
+  mkdir -p "$VENDOR_DIR/$VENDOR_NAME"
+  chdir "$VENDOR_DIR/$VENDOR_NAME"
+  [[ -n "$HOMEBREW_QUIET" ]] || echo "==> Installing $(basename "$VENDOR_URL")"
+  trap '' SIGINT
+  tar "$tar_args" "$CACHED_LOCATION"
+  if "./$VENDOR_VERSION/bin/$VENDOR_NAME" --version >/dev/null 2>&1
+  then
+    ln -sfn "$VENDOR_VERSION" opt
+    # remove old vendor installations by sorting files with modified time.
+    ls -t | grep -Ev "^(opt|$VENDOR_VERSION)" | tail -n +4 | xargs rm -rf
+  else
+    rm -rf "$VENDOR_VERSION"
+    [[ -n "$HOMEBREW_QUIET" ]] || onoe "Installing vendor $VENDOR_NAME failed."
+  fi
+  trap - SIGINT
+}
+
+homebrew-install-vendor() {
+  local option
+  local url_var
+  local sha_var
+
+  for option in "$@"
+  do
+    case "$option" in
+      -\?|-h|--help|--usage) brew help install-vendor; exit $? ;;
+      --verbose) HOMEBREW_VERBOSE=1 ;;
+      --quiet) HOMEBREW_QUIET=1 ;;
+      --debug) HOMEBREW_DEBUG=1 ;;
+      --*) ;;
+      -*)
+        [[ "$option" = *v* ]] && HOMEBREW_VERBOSE=1
+        [[ "$option" = *q* ]] && HOMEBREW_QUIET=1
+        [[ "$option" = *d* ]] && HOMEBREW_DEBUG=1
+        ;;
+      *)
+        [[ -n "$VENDOR_NAME" ]] && odie "This command does not take multiple vendor targets"
```

Comment:
One ore place that might benefit from using `[[ <inverse-condition> ]] ||` instead of `[[ <condition> ]] &&`.